xdem/spatial_tools.py

```diff
@@ -396,3 +396,42 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # Return the hillshade, scaled to uint8 ranges.
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
+
+
+def subsample_raster(array: Union[np.ndarray, np.ma.masked_array], subsample: Union[float, int],
+                     return_indices: bool = False) -> np.ndarray:
+    """
+    Randomly subsample a 1D or 2D array by a subsampling factor, taking only non NaN/masked values.
+
+    :param subsample: If < 1, will be considered a fraction of valid pixels to extract.
```

Comment:
If <= 1 (to be consistent with the functionality)

Reply:
Good catch!

Reply:
Done in 05f5e40